### PR TITLE
Support Docker using docker-tramp

### PR DESCRIPTION
**Pull request by norci**
for #65 .
In the docker container, the listen address should be 0.0.0.0, not localhost, or port can not be published.

### CODE REVIEW

**Comment by norci** [comment]
I used [docker-tramp](https://github.com/emacs-pe/docker-tramp.el) to open files. then julia snail worked perfectly.

I added these custom vars:
```
(custom-set-variables
 '(julia-snail-host "0.0.0.0")
)
```
added a file in the docker compose dir.
`docker-compose.override.yml`
```
version: "3"
services:
  julia:
    ports:
      - 127.0.0.1:10011:10011
    volumes:
      - ${HOME}/.emacs.d/julia-snail/:${HOME}/.emacs.d/julia-snail/:ro
```

I want to make `julia-snail-host` a local var, if the tramp method is "docker", then set its value to "0". But I have no idea about it. Could you tell me how? thanks.

**Comment by gcv** [comment]
Nice! Please post a complete sample docker compose configuration so I can give this a try.

**Comment by norci** [comment]
try this
run
`
docker run --name julia-1 --rm -it -p 127.0.0.1:10011:10011 -v ${HOME}/.emacs.d/julia-snail/:${HOME}/.emacs.d/julia-snail/ julia bash
`
then eval
`(docker-container-find-directory "julia-1" "/")`
in buffer `/docker:julia-1:/` run `julia-snail`.

File: julia-snail.el [hunk 1]
```diff
@@ -492,7 +508,7 @@ returns \"/home/username/file.jl\"."
           (user-error "The vterm buffer is inactive; double-check julia-snail-executable path"))
         ;; now try to send the Snail startup command
         (julia-snail--send-to-repl
-          (format "JuliaSnail.start(%d); # please wait, time-to-first-plot..." (or julia-snail-remote-port julia-snail-port))
+          (format "JuliaSnail.start(%d; addr=\"%s\"); # please wait, time-to-first-plot..." (or julia-snail-remote-port julia-snail-port) julia-snail-host)
```
Comment (gcv) [inline julia-snail.el:511]: What if we do this instead?

```elisp
(format "JuliaSnail.start(%d%s); # please wait, time-to-first-plot..."
        (or julia-snail-remote-port julia-snail-port)
        (if (string-equal "docker" (file-remote-p (buffer-file-name julia-snail--repl-go-back-target) 'method))
            "; addr=\"0.0.0.0\""
          ""))
```

I didn't test very thoroughly, but it seems to work. It lets us eliminate the `julia-snail-host` variable altogether, though perhaps it's valuable for other use cases?

**Review by gcv — CHANGES_REQUESTED** [review]
This is really good! Exciting to have container Julia support added to Snail. Please go over in-line comments.

**Comment by norci** [comment]
thanks for your advise.
I have tested it on local, ssh, docker. it worked.

**Comment by gcv** [comment]
Thank you for your contribution to Snail! I merged your PR, and added documentation to the CHANGELOG and README files.

**Pull request merged**